packages/compass-explain-plan/src/modules/explain.js

```diff
@@ -320,16 +348,28 @@ export const fetchExplainPlan = (query) => {
       return;
     }
 
+    // Cancel previous run of aggregation if there is any
+    explain.abortController?.abort();
+
+    const abortController = new AbortController();
+    const abortSignal = abortController.signal;
+    const oldExplain = { ...explain };
+    dispatch(startExplainPlan(abortController, oldExplain));
```

Comment:
You can directly dispatch `explainStateChanged` here or simply 
```js
{
  type: EXPLAIN_STATE_CHANGED,
  explainState,
  abortController,
  oldExplain,
}```

Currently `startExplainPlan` calls `explainStateChanged`, which does the same thing. So you can avoid this and directly dispatch the action. 
Also in `doChangeExplainPlanState`, you can access the `oldExplain` from the state and don't have to pass it along.

Reply:
thanks, it definitely make sense. changes implemented

Reply:
ahh wait just to clarify - I have kept `explainStateChanged` as it was used in components and to respond to events on registry. For them it would still make sense to keep it as it makes it more clear what is happening